agent/qrexec-agent-data.c

```diff
@@ -632,23 +636,30 @@ static int handle_new_process_common(int type, int connect_domain, int connect_p
         case MSG_JUST_EXEC:
             send_exit_code(data_vchan, handle_just_exec(cmdline));
             break;
-        case MSG_EXEC_CMDLINE:
-            if (execute_qubes_rpc_command(cmdline, &pid, &stdin_fd, &stdout_fd, &stderr_fd, !qrexec_is_fork_server) < 0)
+        case MSG_EXEC_CMDLINE: {
+            struct buffer stdin_buf;
```

Comment:
Please initialize the buffer immediately.

(In general I'm not too happy with this stdin_buf solution but currently don't have a better idea)